src/main/java/com/klodnicki/bike/rest/API/Bike/rest/controller/BikeController.java

```diff
@@ -36,12 +41,38 @@ public void deleteBikeById(@PathVariable("id") Long id) throws NotFoundInDatabas
         bikeService.deleteById(id);
     }
 
-    @PutMapping("/{id}")
-    public Bike updateBike(@PathVariable Long id, @RequestBody @Valid Bike bikeToUpdate) throws NotFoundInDatabaseException {
-        return bikeService.updateBike(id, bikeToUpdate);
+    @PutMapping("/admin/{id}")
+    public Bike adminUpdateBike(@PathVariable Long id, @RequestBody @Valid Bike bikeToUpdate,
+           @AuthenticationPrincipal UserDetails user) throws NotFoundInDatabaseException, UnauthorizedException {
+
+//        if (user.getUsername().equals("admin")) {
+//            return bikeService.updateBikeAdmin(id, bikeToUpdate);
+//        }
+
+        for (GrantedAuthority ga : user.getAuthorities()) {
+            if (ga.getAuthority().equals("ROLE_ADMIN")) {
+                return bikeService.updateBikeAdmin(id, bikeToUpdate);
+            }
+        }
+
+        throw new UnauthorizedException();
     }
 
+    @PutMapping("/user/{id}")
+    public BikeForNormalUserDTO userUpdateBike(@PathVariable Long id, @RequestBody @Valid Bike bikeToUpdate,
+       @AuthenticationPrincipal UserDetails user) throws UnauthorizedException, NotFoundInDatabaseException {
 
+//        if(user.getUsername().equals("user")) {
+//            return bikeService.updateBikeUser(id, bikeToUpdate);
+//        }
```

Comment:
Keeping commented code is a bad practice!

Reply:
I know its only for a practice workshop